src/NodeRTLib/ProjectFiles/NodeRtUtils.cpp

```diff
@@ -168,11 +177,22 @@ namespace NodeRT { namespace Utils {
   // compares 2 strings using a case insensitive comparison
   bool CaseInsenstiveEquals(const wchar_t* str1, const uint16_t* str2)
   {
-    int maxCount = static_cast<int>(min(wcslen(str1), wcslen(reinterpret_cast<const wchar_t*>(str2))));
-    return (_wcsnicmp(str1, reinterpret_cast<const wchar_t*>(str2), maxCount) == 0);
+	const wchar_t* str2Casted = reinterpret_cast<const wchar_t*>(str2);
+
+	int maxCount = static_cast<int>(min(wcslen(str1), wcslen(str2Casted)));
+	return (_wcsnicmp(str1, str2Casted, maxCount) == 0);
   }
 #endif
 
+  bool CaseInsenstiveEquals(const char* str1, const uint16_t* str2)
+  {
+	  const char* str2Casted = reinterpret_cast<const char*>(str2);
```

Comment:
The comparison here is made between a unicode (2 byte) string and an ascii string, and since the 2 byte one is forced to act as a single byte, it will result with the comparison to be made to the second byte as well.
I think that we should always compare the same type to the same type.
In case we compare unicode string from the WinRT world, always compare them to unicode - that should be fine. (If we are converting them from the "node" world, we can use the proper String::Value to get the type that we want, e.g. String::Value for unicode and String::UTF8Value for single byte)